website/project/model.py

```diff
@@ -2421,6 +2426,18 @@ def serialize(self):
             'is_registration': self.is_registration
         }
 
+    def retract_registration(self, user, justification=None):
+
+        if not self.is_public or not self.is_registration:
+            raise ValidationTypeError
+
+        if justification:
+            self.retracted_justification = justification
+        self.is_retracted = True
+        self.retraction_date = datetime.datetime.utcnow()
+        self.retracted_by = user
+
+        self.save()
```

Comment:
Avoid implicit calls to `save()`--they often add a lot of unnecessary computation when doing a batch of updates.